src/mosquito/runners/overseer.cr

```diff
@@ -0,0 +1,64 @@
+module Mosquito::Runners
+  # The Overseer is responsible for managing:
+  # - a `Coordinator`
+  # - an `Executor`
+  # - the `QueueList`
+  # - any idle state as configured
+  #
+  # An overseer manages the loop that each thread or process runs.
+  class Overseer
+    include RunAtMost
+
+    Log = ::Log.for self
+
+    # Minimum time in seconds to wait between checking for jobs.
+    property idle_wait : Time::Span {
+      Mosquito.configuration.idle_wait
+    }
+
+    property keep_running : Bool
+
+    getter queue_list, executor, coordinator
+
+    def initialize
+      @queue_list = QueueList.new
+      @coordinator = Coordinator.new queue_list
+      @executor = Executor.new queue_list
+
+      @keep_running = true
+    end
+
+    def worker_id
+      "Worker [#{coordinator.instance_id}]"
+    end
+
+    def stop
+      Log.info { worker_id + " is done after this job." }
```

Comment:
```suggestion
      Log.info { "#{worker_id} is done after this job." }
```
Super minor, but is a good practice in general to avoid intermediate strings